### PR TITLE
Bluetooth: L2CAP improvements

### PR DESCRIPTION
**Pull request by Vudentz**
A few improvement to L2CAP layer:

1. Move fixed channels to ROM: Similarly to what has been done to static GATT services.
2. Increase net_buf user_data to 8 bytes so it can store a user_data in addition to a callback then use it to implement a sent callback to notify when a SDU has been completely sent per channel.
3. Add a status callback to notify when channel status changes e.g. when more credits has been added notify out status.

### CODE REVIEW

File: subsys/net/Kconfig [hunk 1]
```diff
@@ -18,8 +18,8 @@ if NET_BUF
 
 config NET_BUF_USER_DATA_SIZE
 	int "Size of user_data available in every network buffer"
-	default 4
-	range 4 65535 if BT
+	default 8
```
Comment (jhedberg) [inline subsys/net/Kconfig]: What's the reason for changing the default for everyone, even those who don't use Bluetooth? I'd do two defaults, just like the range:
```
default 8 if BT
default 4
```

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: subsys/net/Kconfig (hunk 1 above)
Comment (Vudentz) [reply subsys/net/Kconfig]: Right, this was intended just to BT actually

**Review by jhedberg — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by Vudentz** [comment]
@jhedberg can we push this one?

**Pull request merged**